app/merginapi.cpp

```diff
@@ -1025,77 +1119,80 @@ ProjectList MerginApi::parseProjectsData( const QByteArray &data, bool dataFromS
+
+  QByteArray data;
+  if ( file.open( QIODevice::ReadOnly ) )
   {
-    return false;
+    data = file.readAll();
+    file.close();
   }
 
-  file.write( data );
-  file.close();
-
-  return true;
+  QJsonDocument doc = QJsonDocument::fromJson( data );
+  QJsonObject projectMap = doc.object();
+
+  MerginProject p;
+  p.name = projectMap.value( QStringLiteral( "name" ) ).toString();
+  p.projectNamespace = projectMap.value( QStringLiteral( "namespace" ) ).toString();
+  p.created = QDateTime::fromString( projectMap.value( QStringLiteral( "created" ) ).toString(), Qt::ISODateWithMs ).toUTC();
+  QDateTime updated = QDateTime::fromString( projectMap.value( QStringLiteral( "updated" ) ).toString(), Qt::ISODateWithMs ).toUTC();
+  if ( !updated.isValid() )
+    updated = p.created;
+  p.updated = updated;
+  p.lastSync = QDateTime::fromString( projectMap.value( QStringLiteral( "lastSync" ) ).toString(), Qt::ISODateWithMs ).toUTC();
+  p.projectDir = projectPath;
+
+  return  std::make_shared<MerginProject>( p );
 }
 
-void MerginApi::cacheProjects()
+// TODO checksums + other data in the next task
+bool MerginApi::cacheProjectData( const QString &projectDir, const QString &projectFullName )
 {
-  QJsonDocument doc;
-  QJsonArray array;
-  for ( std::shared_ptr<MerginProject> p : mMerginProjects )
+  std::shared_ptr<MerginProject> project = getProject( projectFullName );
+  if ( project )
   {
-//      QDir projectFolder( mDataDir + '/' + p->name );
-//      if ( !projectFolder.exists() )
-//      {
-//          continue;
-//      }
-
+    QJsonDocument doc;
     QJsonObject projectMap;
-    projectMap.insert( QStringLiteral( "created" ), p->created.toString( Qt::ISODateWithMs ) );
-    projectMap.insert( QStringLiteral( "updated" ), p->updated.toString( Qt::ISODateWithMs ) );
-    projectMap.insert( QStringLiteral( "lastSync" ), p->lastSync.toString( Qt::ISODateWithMs ) );
-    projectMap.insert( QStringLiteral( "creator" ), p->creator );
-    projectMap.insert( QStringLiteral( "name" ), p->name );
-    projectMap.insert( QStringLiteral( "namespace" ), p->projectNamespace );
-
-    QJsonArray tags;
-    projectMap.insert( QStringLiteral( "tags" ), tags.fromStringList( p->tags ) );
-    QJsonArray writers;
-    for ( int userId : p->writers )
-    {
-      writers.append( userId );
-    }
-    projectMap.insert( QStringLiteral( "writers" ), writers );
-    array.append( projectMap );
+    projectMap.insert( QStringLiteral( "created" ), project->created.toString( Qt::ISODateWithMs ) );
+    projectMap.insert( QStringLiteral( "updated" ), project->updated.toString( Qt::ISODateWithMs ) );
+    projectMap.insert( QStringLiteral( "lastSync" ), project->lastSync.toString( Qt::ISODateWithMs ) );
+    projectMap.insert( QStringLiteral( "name" ), project->name );
+    projectMap.insert( QStringLiteral( "namespace" ), project->projectNamespace );
+
+    doc.setObject( projectMap );
+    cacheData( doc.toJson(), projectDir + "/" + mCacheFile );
```

Comment:
can we rename mCacheFile to something like mMetadataFile ? (or with "s" prefix instead of "m" if it will be static)
That file does not really cache anything, just keeps project metadata - best to avoid possible confusion